ingress/controllers/nginx/configuration.md

```diff
@@ -0,0 +1,339 @@
+
+**Please note the template is tied to the go code. Be sure to no change names in the variable `$cfg`**
+
+
+### Custom NGINX upstream checks
+
+NGINX exposes some flags in the [upstream configuration](http://nginx.org/en/docs/http/ngx_http_upstream_module.html#upstream) that enables the configuration of each server in the upstream. The ingress controller allows custom `max_fails` and `fail_timeout` parameters in a global context using `upstream-max-fails` or `upstream-fail-timeout` in the NGINX config map or in a particular Ingress rule. By default this values are 0. This means NGINX will respect the `readinessProbe`, if is defined. If there is no probe, NGINX will not mark a server inside an upstream down.
+
+**With the default values NGINX will not health check your backends, and whenever the endpoints controller notices a readiness probe failure that pod's ip will be removed from the list of endpoints, causing nginx to also remove it from the upstreams.**
+
+To use custom values in an Ingress rule define this annotations:
+
+`ingress.kubernetes.io/upstream-max-fails`: number of unsuccessful attempts to communicate with the server that should happen in the duration set by the fail_timeout parameter to consider the server unavailable
+
+`ingress.kubernetes.io/upstream-fail-timeout`: time in seconds during which the specified number of unsuccessful attempts to communicate with the server should happen to consider the server unavailable. Also the period of time the server will be considered unavailable.
+
+**Important:** 
+The upstreams are shared. i.e. Ingress rule using the same service will use the same upstream. 
+This means only one of the rules should define annotations to configure the upstream servers
+
+
+Please check the [custom upstream check](examples/custom-upstream-check/README.md) example
+
+
+### Authentication
+
+Is possible to add authentication adding additional annotations in the Ingress rule. The source of the authentication is a secret that contains usernames and passwords inside the the key `auth`
+
+The annotations are:
+
+```
+ingress.kubernetes.io/auth-type:[basic|digest]
+```
+
+Indicates the [HTTP Authentication Type: Basic or Digest Access Authentication](https://tools.ietf.org/html/rfc2617).
+
+```
+ingress.kubernetes.io/auth-secret:secretName
+```
+
+Name of the secret that contains the usernames and passwords with access to the `path/s` defined in the Ingress Rule.
+The secret must be created in the same namespace than the Ingress rule
+
+```
+ingress.kubernetes.io/auth-realm:"realm string"
+```
+
+Please check the [auth](examples/custom-upstream-check/README.md) example
+
+
+### Rewrite
+
+In some scenarios the exposed URL in the backend service differs from the specified path in the Ingress rule. Without a rewrite any request will return 404.
+To allow this is possible to use the annotation `ingress.kubernetes.io/rewrite-target` with the path expected in the service.
```

Comment:
set the annotation `ingress..` to the path expected by the service

actually this bit is a little confusing, what's the source url? if it doesn't work with multiple source urls, plese mentione that